### PR TITLE
GTC-2794: Only owner or admin can use mutate dataset endpoints

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [ ] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: app/routes/datasets/dataset.py [hunk 1]
```diff
@@ -25,6 +25,16 @@
 router = APIRouter()
 
 
+async def get_owner(
+    dataset: str = Depends(dataset_dependency), user: User = Depends(get_manager)
+) -> User:
+    dataset_row: ORMDataset = await datasets.get_dataset(dataset)
```
Comment (danscales) [inline app/routes/datasets/dataset.py:31]: Can you add a """ .... """ comment line describing the function above this line?

**Review by danscales — APPROVED** [review]
Will be easier to read once Justin's changes are merged into feature/data_manager (since this current diff includes all of Justin's changes), but yes, your changes look good to me.  Thanks for writing the fail and success test cases.

I assume you purposely skipped adding a test for update_dataset, since it is so similar to delete_dataset?  Seems ok to not add tests for update_dataset.

File: app/routes/datasets/dataset.py (hunk 1 above)
Comment (dmannarino) [reply app/routes/datasets/dataset.py:31]: Will do, thanks! Also, I'm adding more tests.

**Review by solomon-negusse — COMMENTED** [review]
Looks good, great test coverage!

**Pull request merged**